terminatorlib/terminal.py

```diff
@@ -1453,12 +1455,15 @@ def spawn_child(self, widget=None, respawn=False, debugserver=False):
             self.set_cwd(options.working_directory)
             options.working_directory = ''
 
-        if type(command) is list:
+        if type(execute) is list:
+            shell = util.shell_lookup()
+            args.insert(0, shell)
+            args += ['-c', execute[0]]
```

Comment:
What about the rest of the elements of the execute list?

-x/--execute uses the rest of the arguments (list, not string) as the command to execute. It is not meant to be a single shell-parsable script. Here, --command is also passed to the shell but other terminal emulators, like gnome-terminal, do not; Instead, they split the command as a shell would and execute it.

I don't think these options should be used to execute commands meant to be interpreted by a shell. Perhaps you should try `terminator -x bash -c "echo yo; sleep 3"`